core/utilities/Logger.cpp

```diff
@@ -18,24 +18,224 @@
 
 #include <algorithm>
 #include <iostream>
+#include <spdlog/common.h>
+#include <spdlog/details/file_helper.h>
+#include <spdlog/sinks/daily_file_sink.h>
+#include <spdlog/sinks/ringbuffer_sink.h>
+#include <spdlog/sinks/stdout_sinks.h>
+
+namespace
+{
+const unsigned BURRER_LOG_SIZE = 5000;
+}
 
 namespace wolkabout
 {
-std::unique_ptr<Logger> Logger::m_instance;
+struct custom_filename_calculator
+{
+    // Create filename for the form basename.YYYY-MM-DD_hh-mm.ext
+    static spdlog::filename_t calc_filename(const spdlog::filename_t& filename, const tm& now_tm)
+    {
+        spdlog::filename_t basename, ext;
+        std::tie(basename, ext) = spdlog::details::file_helper::split_by_extension(filename);
+
+        return fmt::format(SPDLOG_FILENAME_T("{}_{:04d}-{:02d}-{:02d}_{:02d}-{:02d}{}"), basename,
+                           now_tm.tm_year + 1900, now_tm.tm_mon + 1, now_tm.tm_mday, now_tm.tm_hour, now_tm.tm_min,
+                           ext);
+    }
+};
 
 void Logger::operator+=(Log& log)
 {
+    if (log.getLogLevel() == LogLevel::OFF)
+        return;
     logEntry(log);
 }
 
-void Logger::setInstance(std::unique_ptr<Logger> instance)
+Logger& Logger::getInstance()
+{
+    static Logger instance;
+    return instance;
+}
+
+void Logger::init(LogLevel level, Type type, const std::string& filePathWithExtension)
+{
+    static bool runOnce = false;
```

Comment:
explain purpose of limiting to one?

it is by default nature of the any init to call it once.

Reply:
Not sure about the exact reason for this, perhaps to stop logger duplication?
Maybe @milanzlicic has better input